engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobYieldProcessor.java

```diff
@@ -17,55 +17,41 @@
 import io.camunda.zeebe.engine.state.immutable.JobState.State;
 import io.camunda.zeebe.engine.state.immutable.ProcessingState;
 import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
-import io.camunda.zeebe.protocol.record.RejectionType;
 import io.camunda.zeebe.protocol.record.intent.JobIntent;
 import io.camunda.zeebe.stream.api.records.TypedRecord;
+import java.util.List;
 
 public final class JobYieldProcessor implements TypedRecordProcessor<JobRecord> {
-
-  public static final String NOT_ACTIVATED_JOB_MESSAGE =
-      "Expected to yield activated job with key '%d', but %s";
   private final JobState jobState;
   private final BpmnJobActivationBehavior jobActivationBehavior;
   private final StateWriter stateWriter;
   private final TypedRejectionWriter rejectionWriter;
+  private final JobCommandPreconditionChecker preconditionChecker;
 
   public JobYieldProcessor(
       final ProcessingState state, final BpmnBehaviors bpmnBehaviors, final Writers writers) {
     jobState = state.getJobState();
     jobActivationBehavior = bpmnBehaviors.jobActivationBehavior();
     stateWriter = writers.state();
     rejectionWriter = writers.rejection();
+    preconditionChecker = new JobCommandPreconditionChecker("yield", List.of(State.ACTIVATED));
   }
 
   @Override
   public void processRecord(final TypedRecord<JobRecord> record) {
-    final long key = record.getKey();
-    final JobState.State state = jobState.getState(key);
-    if (state == State.ACTIVATED) {
-      final JobRecord yieldedJob = record.getValue();
-
-      stateWriter.appendFollowUpEvent(key, JobIntent.YIELDED, yieldedJob);
-      jobActivationBehavior.notifyJobAvailableAsSideEffect(yieldedJob);
-    } else {
-      final String textState;
-
-      switch (state) {
-        case ACTIVATABLE:
-          textState = "it must be activated first";
-          break;
-        case FAILED:
-          textState = "it is marked as failed";
-          break;
-        default:
-          textState = "no such job was found";
-          break;
```

Comment:
:x: Same as above, let's keep the error messages as they are.

Since this processor is new, you can align the error messages with the ones from the existing Processor classes.

Maybe in the `JobCommandPreconditionChecker` we can keep a `Map` of `JobState` keys and default `error_message` values, and override them if they are different in the processor.